apps/channels/models.py

```diff
@@ -34,18 +37,30 @@ def form(self):
         return ChannelForm(initial={"platform": self})
 
     def extra_form(self, *args, **kwargs):
-        from apps.channels.forms import TelegramChannelForm, WhatsappChannelForm
+        from apps.channels import forms
 
         match self:
             case self.TELEGRAM:
-                return TelegramChannelForm(*args, **kwargs)
+                return forms.TelegramChannelForm(*args, **kwargs)
             case self.WHATSAPP:
-                return WhatsappChannelForm(*args, **kwargs)
+                return forms.WhatsappChannelForm(*args, **kwargs)
+            case self.FACEBOOK:
+                initial = kwargs.get("initial", {})
+                initial.setdefault("verify_token", str(uuid.uuid4()))
+                kwargs["initial"] = initial
+                return forms.FacebookChannelForm(*args, **kwargs)
+
+
+class ExperimentChannelObjectManager(models.Manager):
+    def filter_extras(self, key: str, value: str, platform: ChannelPlatform, team: str):
+        extra_data_filter = Q(extra_data__contains={key: value})
+        return super().get_queryset().filter(extra_data_filter).filter(experiment__team__slug=team, platform=platform)
```

Comment:
nit: you don't need to call `super().get_queryset()`
```suggestion
        return self.filter(extra_data_filter).filter(experiment__team__slug=team, platform=platform)
```

Reply:
https://github.com/dimagi/open-chat-studio/pull/115/commits/ce2298bedab43b56ec9d9aeb6e4fb6a1a217534e